test/files/scalacheck/MutableTreeSet.scala

```diff
@@ -0,0 +1,200 @@
+import java.io._
+
+import org.scalacheck._
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Prop.forAll
+
+import scala.collection.generic.CanBuildFrom
+import scala.collection.mutable
+import scala.util.Try
+
+package scala.collection.mutable {
+
+  object MutableTreeSetProperties extends Properties("mutable.TreeSet") {
+    type K = String
+
+    property("size, isEmpty") = forAll { (elems: Set[K]) =>
+      val set = mutable.TreeSet[K]()
+      set ++= elems
+      set.size == elems.size && set.isEmpty == elems.isEmpty
+    }
+
+    property("+=") = forAll { (set: mutable.TreeSet[K], k: K) =>
+      val oldSize = set.size
+      val containedKeyBefore = set.contains(k)
+      val newExpectedSize = if(containedKeyBefore) oldSize else oldSize + 1
+
+      set += k
+      set.contains(k) && set.size == newExpectedSize
+    }
+
+    property("++=") = forAll { (set: mutable.TreeSet[K], ks: Seq[K]) =>
+      set ++= ks
+      ks.toSet.forall(set.contains)
```

Comment:
You should also check that the `set` still contains the elements it originally contained. Tree manipulations are prone to dropping nodes.